### PR TITLE
Noodlepad Additions and Updates

### PR DESCRIPTION
**Pull request by jessel92**
Modified this keyboard according to failed PR from the previous day with help from @sigprof

## Description

Renaming `themadnoodle/noodlepad`to `themadnoodle/noodlepad/v1` and added `themadnoodle/noodlepad/v2` 
Updated the formatting to the current JSON QMK formatting to the best of my abilities. 
Cleaned up keymaps and readmes

Got a HUGE amount of help from @sigprof via QMK Discord, so thank you so much! 

I also have other PRs Open for:

Adding Keyboard The Mad Noodle Udon13 #22704 
Added Keyboard Noodlepad_Micro #22703

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/themadnoodle/noodlepad/config.h [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2023 Jesse Leventhal (@The=Mad-Noodle)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+/*RGB Defaults*/
+#define RGBLIGHT_DEFAULT_MODE RGBLIGHT_MODE_RAINBOW_SWIRL + 5
+/* See info.json for more RGB settings*/
```
Comment (waffle87) [inline keyboards/themadnoodle/noodlepad/config.h:8]:
```suggestion
#define RGBLIGHT_DEFAULT_MODE RGBLIGHT_MODE_RAINBOW_SWIRL + 5
```

File: keyboards/themadnoodle/noodlepad/v1/readme.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# NoodlePad v.1
+
+![NoodPad v1](https://static.wixstatic.com/media/59d0ff_a3131781ceb744758789eccec076c3e0~mv2.jpg)
```
Comment (waffle87) [inline keyboards/themadnoodle/noodlepad/v1/readme.md:3]: Please rehost this image on Imgur.

File: keyboards/themadnoodle/noodlepad/v2/info.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    },
+    "usb": {
+        "device_version": "2.0.0",
+        "pid": "0x0002",
+        "vid": "0x6A6C"
+    },
+    "encoder": {
+        "enabled": true,
+        "rotary": [
+            { "pin_a": "D0", "pin_b": "D1", "resolution": 4 }
```
Comment (waffle87) [inline keyboards/themadnoodle/noodlepad/v2/info.json:21]:
```suggestion
        "rotary": [
            { "pin_a": "D0", "pin_b": "D1" }
```

File: keyboards/themadnoodle/noodlepad/v1/info.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    "processor": "atmega32u4",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0001",
+        "vid": "0x6A6C"
+    },
+    "encoder": {
+        "enabled": true,
+        "rotary": [
+            { "pin_a": "D0", "pin_b": "D1", "resolution": 4 }
```
Comment (waffle87) [inline keyboards/themadnoodle/noodlepad/v1/info.json:22]:
```suggestion
        "rotary": [
            { "pin_a": "D0", "pin_b": "D1" }
```

File: keyboards/themadnoodle/noodlepad/v2/readme.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# NoodlePad v.2
+
+![NoodPad v2](https://static.wixstatic.com/media/59d0ff_e740020d15824bf59b7509bdd5178188~mv2.jpg)
```
Comment (waffle87) [inline keyboards/themadnoodle/noodlepad/v2/readme.md:3]: Please rehost this image on Imgur.

File: keyboards/themadnoodle/noodlepad/info.json [hunk 6]
```diff
@@ -1,59 +1,51 @@
...
+    "processor": "atmega32u4",
+    "diode_direction": "ROW2COL",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgblight": true
```
Comment (waffle87) [inline keyboards/themadnoodle/noodlepad/info.json:16]:
```suggestion
        "rgblight": true,
        "encoder": true
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/themadnoodle/noodlepad/v1/readme.md (hunk 2 above)
Comment (jessel92) [reply keyboards/themadnoodle/noodlepad/v1/readme.md]: Don

File: keyboards/themadnoodle/noodlepad/v2/readme.md (hunk 5 above)
Comment (jessel92) [reply keyboards/themadnoodle/noodlepad/v2/readme.md]: and done =)

**Review by jessel92 — COMMENTED** [review]
All should be good to go

**Comment by zvecr** [comment]
> All should be good to go

image

Nope, thats a hard blocker on the PR being mergable.

**Comment by jessel92** [comment]
> > All should be good to go
> 
> image
> 
> Nope, thats a hard blocker on the PR being mergable.

What does that mean? What caused it?

**Comment by zvecr** [comment]
lib changes in [fb05cce](https://github.com/qmk/qmk_firmware/pull/22701/commits/fb05cceea6a12dfeccf258fc75f9445a35694611)

**Comment by adophoxia** [comment]
There is [`lib/chibios-contrib`](https://github.com/qmk/qmk_firmware/pull/22701/files#diff-128d5d8f6685c55d3fdcf4c4cacd61d69041c9f753c54a943f54a67a06a7990e) submodule changes in this PR.

**Comment by jessel92** [comment]
> There is [`lib/chibios-contrib`](https://github.com/qmk/qmk_firmware/pull/22701/files#diff-128d5d8f6685c55d3fdcf4c4cacd61d69041c9f753c54a943f54a67a06a7990e) submodule changes in this PR. Most likely snuck in while you were making changes previously.

Well crap, can I do something about it???

**Comment by jessel92** [comment]
@zvecr @adophoxia I reverted the commit. Should that fix it?

**Comment by jessel92** [comment]
> @zvecr @adophoxia I reverted the commit. Should that fix it?

IT FIXED IT!

**Comment by jessel92** [comment]
> @zvecr @adophoxia I reverted the commit. Should that fix it?

@zvecr @adophoxia  IT FIXED IT! We should be good now right?

File: keyboards/themadnoodle/noodlepad/v2/readme.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# NoodlePad v.2
+
+![NoodPad v2](https://i.imgur.com/tzP0tdz.jpg)
```
Comment (waffle87) [inline keyboards/themadnoodle/noodlepad/v2/readme.md:3]:
```suggestion
![NoodPad v2](https://i.imgur.com/tzP0tdzh.jpg)
```

**Review by waffle87 — APPROVED** [review]

File: keyboards/themadnoodle/noodlepad/v2/info.json [hunk 8]
```diff
@@ -0,0 +1,24 @@
+{
+    "manufacturer": "The Mad Noodle",
+    "keyboard_name": "NoodlePad v2",
+    "maintainer": "The-Mad-Noodle",
+    "url": "https://www.madnoodleprototypes.com/",
```
Comment (fauxpark) [inline keyboards/themadnoodle/noodlepad/v2/info.json:5]:
```suggestion
    "url": "https://www.madnoodleprototypes.com/",
```

File: keyboards/themadnoodle/noodlepad/v1/readme.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
+# NoodlePad v.1
+
+![NoodPad v1](https://i.imgur.com/dmThaHq.jpg)
+
+*The Noodlepad v.1 is a 8 key macro keypad with a push button encoder knob*
```
Comment (fauxpark) [inline keyboards/themadnoodle/noodlepad/v1/readme.md:5]:
```suggestion
The Noodlepad v.1 is a 8 key macro keypad with a push button encoder knob.
```

File: keyboards/themadnoodle/noodlepad/v2/readme.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
+# NoodlePad v.2
+
+![NoodPad v2](https://i.imgur.com/tzP0tdzh.jpg)
+
+*The Noodlepad v.2 is a 8 hotswap key macro keypad with a push button encoder knob*
```
Comment (fauxpark) [inline keyboards/themadnoodle/noodlepad/v2/readme.md:5]:
```suggestion
The Noodlepad v.2 is a 8 hotswap key macro keypad with a push button encoder knob.
```

File: keyboards/themadnoodle/noodlepad/keymaps/via/rules.mk [hunk 11]
```diff
@@ -0,0 +1,3 @@
+# This file is used to override the default rules.mk file
```
Comment (fauxpark) [inline keyboards/themadnoodle/noodlepad/keymaps/via/rules.mk:1]:
```suggestion
```

File: keyboards/themadnoodle/noodlepad/keymaps/default/rules.mk [hunk 12]
```diff
@@ -0,0 +1,2 @@
+#Keymap specific options
```
Comment (fauxpark) [inline keyboards/themadnoodle/noodlepad/keymaps/default/rules.mk:1]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**